### ISSUE TITLE
Error launching the office.launch.xml [No namespace found]

### ISSUE THREAD

**Issue by KaeYan** [osrf/rmf_demos#57]
Hi, 

After I followed the instruction to install, I encountered the error:

kaeyan@kaeyan1804:~/rmf_demos_ws$ ros2 launch demos office.launch.xml
[INFO] [launch]: All log files can be found below /home/kaeyan/.ros/log/2020-04-02-15-27-48-892762-kaeyan1804-6908
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [rmf_traffic_schedule-1]: process started with pid [6919]
[INFO] [building_map_server-2]: process started with pid [6920]
[INFO] [rviz2-3]: process started with pid [6921]
[INFO] [fleet_state_visualizer-4]: process started with pid [6922]
[INFO] [rviz2-5]: process started with pid [6923]
[INFO] [door_supervisor-6]: process started with pid [6924]
[INFO] [gzserver-7]: process started with pid [6931]
[INFO] [gzclient-8]: process started with pid [6932]
[INFO] [magni_fleet_adapter-9]: process started with pid [6942]
[INFO] [magni_state_aggregator-10]: process started with pid [6954]
[INFO] [dispenser-11]: process started with pid [6957]
[INFO] [dispenser-12]: process started with pid [6958]
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Launching fleet adapter for [magni]
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [linear_velocity] set to: 0.400000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [angular_velocity] set to: 2.200000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [linear_acceleration] set to: 3.000000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [angular_acceleration] set to: 5.000000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [profile_radius] set to: 0.300000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [reversible] set to: 1
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [delay_threshold] set to: 15.000000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [retry_wait] set to: 10.000000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [planning_timeout] set to: 5.000000
[magni_fleet_adapter-9] Adding waypoint [0] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [3] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [5] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [7] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [10] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [13] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [14] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [15] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [16] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [17] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [18] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [19] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [23] as a parking spot
[magni_fleet_adapter-9] Named waypoints:
[magni_fleet_adapter-9]  -- [supplies]
[magni_fleet_adapter-9]  -- [magni2_charger]
[magni_fleet_adapter-9]  -- [coe]
[magni_fleet_adapter-9]  -- [magni1_charger]
[magni_fleet_adapter-9]  -- [hardware_2]
[magni_fleet_adapter-9]  -- [cubicle_2]
[magni_fleet_adapter-9]  -- [pantry]
[magni_fleet_adapter-9]  -- [station_1]
[magni_fleet_adapter-9]  -- [lounge]
[magni_fleet_adapter-9]  -- [cubicle_1]
[magni_fleet_adapter-9]  -- [hardware_1]
[magni_fleet_adapter-9]  -- [station_2]
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [discovery_timeout] set to: 60.000000
[magni_fleet_adapter-9] Starting [magni]
[gzclient-8] Gazebo multi-robot simulator, version 9.0.0
[gzclient-8] Copyright (C) 2012 Open Source Robotics Foundation.
[gzclient-8] Released under the Apache 2 License.
[gzclient-8] http://gazebosim.org
[gzclient-8] 
[gzserver-7] Gazebo multi-robot simulator, version 9.0.0
[gzserver-7] Copyright (C) 2012 Open Source Robotics Foundation.
[gzserver-7] Released under the Apache 2 License.
[gzserver-7] http://gazebosim.org
[gzserver-7] 
[gzserver-7] terminate called after throwing an instance of 'std::length_error'
[gzserver-7]   what():  basic_string::_M_create
[gzclient-8] [Msg] Waiting for master.
[gzclient-8] [Msg] Connected to gazebo master @ http://127.0.0.1:11345
[gzclient-8] [Msg] Publicized address: 192.168.21.197
[gzclient-8] [Err] [Node.cc:105] No namespace found
[ERROR] [gzserver-7]: process has died [pid 6931, exit code -6, cmd 'gzserver --verbose -s libgazebo_ros_factory.so -s libgazebo_ros_init.so /home/kaeyan/rmf_demos_ws/install/rmf_demo_maps/share/rmf_demo_maps/maps/office/office.world'].
[gzclient-8] [Err] [TransportIface.cc:396] Unable to connect to master.
[gzclient-8] [Err] [GLWidget.cc:881] Unable to connect to a running Gazebo master.
[gzclient-8] [Wrn] [Publisher.cc:141] Queue limit reached for topic /gazebo/world/user_camera/pose, deleting message. This warning is printed only once.

When i rosdep install, I actually need to install those separately:
$ sudo apt install -y python-shapely
$ sudo apt install ros-eloquent-gazebo*
$ sudo apt install ros-eloquent-launch*

The build warning is as such:

kaeyan@kaeyan1804:~/rmf_demos_ws$ colcon build
Starting >>> rmf_fleet_msgs
Starting >>> building_map_msgs
Starting >>> rmf_dispenser_msgs
Starting >>> rmf_traffic_msgs
Starting >>> rmf_utils
Starting >>> rmf_schedule_visualizer_msgs
Starting >>> rmf_door_msgs
Starting >>> rmf_lift_msgs
Finished <<< rmf_utils [5.20s]                                         
Starting >>> rmf_traffic                                               
Finished <<< rmf_schedule_visualizer_msgs [24.0s]                         
Starting >>> rviz2_plugin
Finished <<< rmf_door_msgs [32.0s]                                        
Starting >>> rmf_demo_assets
Finished <<< rmf_lift_msgs [35.3s]                                        
Starting >>> rmf_workcell_msgs
Finished <<< rmf_demo_assets [5.06s]                                       
Starting >>> traffic_editor
Finished <<< rmf_dispenser_msgs [37.3s]                                    
Starting >>> rmf_task_msgs
Finished <<< rmf_fleet_msgs [39.7s]                                          
Starting >>> building_gazebo_plugins
Finished <<< rviz2_plugin [28.6s]                                             
Finished <<< building_map_msgs [53.7s]                                         
Starting >>> building_map_tools
Starting >>> rmf_gazebo_plugins
Finished <<< building_map_tools [1.57s]                                     
Starting >>> rmf_demo_maps
[Processing: building_gazebo_plugins, rmf_demo_maps, rmf_gazebo_plugins, rmf_task_msgs, rmf_traffic, rmf_traffic_msgs, rmf_workcell_msgs, traffic_editor]
Finished <<< rmf_workcell_msgs [1min 0s]     
Starting >>> test_maps
Finished <<< test_maps [10.5s]               
Finished <<< rmf_task_msgs [1min 10s]        
Starting >>> rmf_demo_tasks
Starting >>> rmf_rviz_plugin
Finished <<< rmf_demo_tasks [1.40s]          
Finished <<< rmf_demo_maps [53.8s]           
--- stderr: rmf_gazebo_plugins               
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_demos/rmf_gazebo_plugins/src/teleport.cpp: In member function ‘void rmf_gazebo_plugins::TeleportPlugin::dispenser_result_cb(rmf_dispenser_msgs::msg::DispenserResult_<std::allocator<void> >::UniquePtr)’:
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_demos/rmf_gazebo_plugins/src/teleport.cpp:85:10: warning: unused variable ‘status’ [-Wunused-variable]
     auto status = msg->status;
          ^~~~~~
---
Finished <<< rmf_gazebo_plugins [57.9s]
Finished <<< building_gazebo_plugins [1min 26s]
Finished <<< rmf_rviz_plugin [19.3s]        
[Processing: rmf_traffic, rmf_traffic_msgs, traffic_editor]
Finished <<< rmf_traffic_msgs [2min 57s]     
Starting >>> fleet_state_visualizer
Finished <<< fleet_state_visualizer [0.77s]  
Finished <<< traffic_editor [2min 22s]       
--- stderr: rmf_traffic                                            
In file included from /usr/include/eigen3/Eigen/LU:34:0,
                 from /usr/include/eigen3/Eigen/Geometry:16,
                 from /home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/include/rmf_traffic/geometry/SimplePolygon.hpp:24,
                 from /home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/src/rmf_traffic/geometry/SimplePolygon.cpp:20:
/usr/include/eigen3/Eigen/src/LU/Determinant.h: In member function ‘bool rmf_traffic::geometry::SimplePolygon::has_self_intersections() const’:
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘*((void*)& M +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h: In member function ‘bool rmf_traffic::geometry::SimplePolygonInternal::check_self_intersections(rmf_traffic::geometry::SimplePolygonInternal::Intersections*) const’:
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘*((void*)& M +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h: In member function ‘virtual rmf_traffic::geometry::CollisionGeometries rmf_traffic::geometry::SimplePolygonInternal::make_fcl() const’:
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘*((void*)& M +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘*((void*)& M +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘output’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘*((void*)& output +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:55: warning: ‘*((void*)& output +8)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:55: warning: ‘*((void*)& output +16)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp: In function ‘void ____C_A_T_C_H____T_E_S_T____78()’:
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp:2349:56: warning: typedef ‘using VehicleTraits = class rmf_traffic::agv::VehicleTraits’ locally defined but not used [-Wunused-local-typedefs]
   using VehicleTraits = rmf_traffic::agv::VehicleTraits;
                                                        ^
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp:2350:44: warning: typedef ‘using Planner = class rmf_traffic::agv::Planner’ locally defined but not used [-Wunused-local-typedefs]
   using Planner = rmf_traffic::agv::Planner;
                                            ^
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp:2351:44: warning: typedef ‘using Duration = std::chrono::nanoseconds’ locally defined but not used [-Wunused-local-typedefs]
   using Duration = std::chrono::nanoseconds;
                                            ^
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp:2352:53: warning: typedef ‘using DetectConflict = class rmf_traffic::DetectConflict’ locally defined but not used [-Wunused-local-typedefs]
   using DetectConflict = rmf_traffic::DetectConflict;
                                                     ^
---
Finished <<< rmf_traffic [3min 1s]
Starting >>> rmf_traffic_ros2
Finished <<< rmf_traffic_ros2 [15.4s]                                
Starting >>> rmf_schedule_visualizer
Starting >>> rmf_fleet_adapter
Finished <<< rmf_schedule_visualizer [28.5s]                                                                                  
Starting >>> visualizer
Finished <<< visualizer [3.06s]                                                                            
--- stderr: rmf_fleet_adapter                                       
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_fleet_adapter/src/full_control/MoveAction.cpp: In member function ‘void rmf_fleet_adapter::full_control::{anonymous}::MoveAction::cancel(std::chrono::nanoseconds)’:
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_fleet_adapter/src/full_control/MoveAction.cpp:1206:40: warning: unused parameter ‘duration’ [-Wunused-parameter]
   void cancel(std::chrono::nanoseconds duration)
                                        ^~~~~~~~
---
Finished <<< rmf_fleet_adapter [46.1s]
Starting >>> demos
Finished <<< demos [1.75s]                         

Summary: 27 packages finished [4min 9s]
  3 packages had stderr output: rmf_fleet_adapter rmf_gazebo_plugins rmf_traffic

**Comment by aaronchongth** [comment]
Sorry your logs are extremely difficult to read, could you put them into markdown code blocks?
```
Like so
```

Please check if it persists, and if it is a duplicate of any previous issues, like this one, https://github.com/osrf/rmf_demos/issues/28

we are using `python3-shapely` which is installed during the first setup step, while `launch-xml` is an issue that has been just resolved https://github.com/osrf/rmf_demos/pull/56

**Comment by KaeYan** [comment]
Hi, sure aaron. After i install and run the office.launch.xml, the error is this:

```
kaeyan@kaeyan1804:~/rmf_demos_ws$ ros2 launch demos office.launch.xml
[INFO] [launch]: All log files can be found below /home/kaeyan/.ros/log/2020-04-02-15-27-48-892762-kaeyan1804-6908
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [rmf_traffic_schedule-1]: process started with pid [6919]
[INFO] [building_map_server-2]: process started with pid [6920]
[INFO] [rviz2-3]: process started with pid [6921]
[INFO] [fleet_state_visualizer-4]: process started with pid [6922]
[INFO] [rviz2-5]: process started with pid [6923]
[INFO] [door_supervisor-6]: process started with pid [6924]
[INFO] [gzserver-7]: process started with pid [6931]
[INFO] [gzclient-8]: process started with pid [6932]
[INFO] [magni_fleet_adapter-9]: process started with pid [6942]
[INFO] [magni_state_aggregator-10]: process started with pid [6954]
[INFO] [dispenser-11]: process started with pid [6957]
[INFO] [dispenser-12]: process started with pid [6958]
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Launching fleet adapter for [magni]
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [linear_velocity] set to: 0.400000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [angular_velocity] set to: 2.200000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [linear_acceleration] set to: 3.000000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [angular_acceleration] set to: 5.000000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [profile_radius] set to: 0.300000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [reversible] set to: 1
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [delay_threshold] set to: 15.000000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [retry_wait] set to: 10.000000
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [planning_timeout] set to: 5.000000
[magni_fleet_adapter-9] Adding waypoint [0] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [3] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [5] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [7] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [10] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [13] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [14] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [15] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [16] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [17] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [18] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [19] as a parking spot
[magni_fleet_adapter-9] Adding waypoint [23] as a parking spot
[magni_fleet_adapter-9] Named waypoints:
[magni_fleet_adapter-9] -- [supplies]
[magni_fleet_adapter-9] -- [magni2_charger]
[magni_fleet_adapter-9] -- [coe]
[magni_fleet_adapter-9] -- [magni1_charger]
[magni_fleet_adapter-9] -- [hardware_2]
[magni_fleet_adapter-9] -- [cubicle_2]
[magni_fleet_adapter-9] -- [pantry]
[magni_fleet_adapter-9] -- [station_1]
[magni_fleet_adapter-9] -- [lounge]
[magni_fleet_adapter-9] -- [cubicle_1]
[magni_fleet_adapter-9] -- [hardware_1]
[magni_fleet_adapter-9] -- [station_2]
[magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [discovery_timeout] set to: 60.000000
[magni_fleet_adapter-9] Starting [magni]
[gzclient-8] Gazebo multi-robot simulator, version 9.0.0
[gzclient-8] Copyright (C) 2012 Open Source Robotics Foundation.
[gzclient-8] Released under the Apache 2 License.
[gzclient-8] http://gazebosim.org
[gzclient-8]
[gzserver-7] Gazebo multi-robot simulator, version 9.0.0
[gzserver-7] Copyright (C) 2012 Open Source Robotics Foundation.
[gzserver-7] Released under the Apache 2 License.
[gzserver-7] http://gazebosim.org
[gzserver-7]
[gzserver-7] terminate called after throwing an instance of 'std::length_error'
[gzserver-7] what(): basic_string::_M_create
[gzclient-8] [Msg] Waiting for master.
[gzclient-8] [Msg] Connected to gazebo master @ http://127.0.0.1:11345
[gzclient-8] [Msg] Publicized address: 192.168.21.197
[gzclient-8] [Err] [Node.cc:105] No namespace found
[ERROR] [gzserver-7]: process has died [pid 6931, exit code -6, cmd 'gzserver --verbose -s libgazebo_ros_factory.so -s libgazebo_ros_init.so /home/kaeyan/rmf_demos_ws/install/rmf_demo_maps/share/rmf_demo_maps/maps/office/office.world'].
[gzclient-8] [Err] [TransportIface.cc:396] Unable to connect to master.
[gzclient-8] [Err] [GLWidget.cc:881] Unable to connect to a running Gazebo master.
[gzclient-8] [Wrn] [Publisher.cc:141] Queue limit reached for topic /gazebo/world/user_camera/pose, deleting message. This warning is printed only once.
```

**Comment by KaeYan** [comment]
When I rosdep install, I actually need to manually install these 3 packages:
```
$ sudo apt install -y python-shapely
$ sudo apt install ros-eloquent-gazebo*
$ sudo apt install ros-eloquent-launch*
```

The build warning after rosdep and manual install is as such:

```
--- stderr: rmf_fleet_adapter
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_fleet_adapter/src/full_control/MoveAction.cpp: In member function ‘void rmf_fleet_adapter::full_control::{anonymous}::MoveAction::cancel(std::chrono::nanoseconds)’:
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_fleet_adapter/src/full_control/MoveAction.cpp:1206:40: warning: unused parameter ‘duration’ [-Wunused-parameter]
void cancel(std::chrono::nanoseconds duration)
^~~~~~~~

--- stderr: rmf_traffic
In file included from /usr/include/eigen3/Eigen/LU:34:0,
from /usr/include/eigen3/Eigen/Geometry:16,
from /home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/include/rmf_traffic/geometry/SimplePolygon.hpp:24,
from /home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/src/rmf_traffic/geometry/SimplePolygon.cpp:20:
/usr/include/eigen3/Eigen/src/LU/Determinant.h: In member function ‘bool rmf_traffic::geometry::SimplePolygon::has_self_intersections() const’:
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘((void)& M +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h: In member function ‘bool rmf_traffic::geometry::SimplePolygonInternal::check_self_intersections(rmf_traffic::geometry::SimplePolygonInternal::Intersections*) const’:
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘((void)& M +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h: In member function ‘virtual rmf_traffic::geometry::CollisionGeometries rmf_traffic::geometry::SimplePolygonInternal::make_fcl() const’:
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘((void)& M +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘((void)& M +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘output’ may be used uninitialized in this function [-Wmaybe-uninitialized]
return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:25: warning: ‘((void)& output +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:55: warning: ‘((void)& output +8)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
return m.coeff(0,0) * m.coeff(1,1) - m.coeff(1,0) * m.coeff(0,1);
/usr/include/eigen3/Eigen/src/LU/Determinant.h:57:55: warning: ‘((void)& output +16)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp: In function ‘void ____C_A_T_C_H____T_E_S_T____78()’:
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp:2349:56: warning: typedef ‘using VehicleTraits = class rmf_traffic::agv::VehicleTraits’ locally defined but not used [-Wunused-local-typedefs]
using VehicleTraits = rmf_traffic::agv::VehicleTraits;
^
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp:2350:44: warning: typedef ‘using Planner = class rmf_traffic::agv::Planner’ locally defined but not used [-Wunused-local-typedefs]
using Planner = rmf_traffic::agv::Planner;
^
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp:2351:44: warning: typedef ‘using Duration = std::chrono::nanoseconds’ locally defined but not used [-Wunused-local-typedefs]
using Duration = std::chrono::nanoseconds;
^
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_core/rmf_traffic/test/unit/agv/test_Planner.cpp:2352:53: warning: typedef ‘using DetectConflict = class rmf_traffic::DetectConflict’ locally defined but not used [-Wunused-local-typedefs]
using DetectConflict = rmf_traffic::DetectConflict;
^

--- stderr: rmf_gazebo_plugins
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_demos/rmf_gazebo_plugins/src/teleport.cpp: In member function ‘void rmf_gazebo_plugins::TeleportPlugin::dispenser_result_cb(rmf_dispenser_msgs::msg::DispenserResult_<std::allocator >::UniquePtr)’:
/home/kaeyan/rmf_demos_ws/src/rmf/rmf_demos/rmf_gazebo_plugins/src/teleport.cpp:85:10: warning: unused variable ‘status’ [-Wunused-variable]
auto status = msg->status;
^~~~~~

```

**Comment by aaronchongth** [comment]
regarding `python-shapely`, since we have been building RMF around ROS2, we will be focusing on only `python3`, hence the installation step for `non-ROS prerequisites` near the top of the README, installs `python3-shapely`.

with the current fixes into `master`, https://github.com/osrf/rmf_demos/blob/master/demos/package.xml#L28, `rosdep` will help install `ros-eloquent-launch-xml`, which should fix any of your launch related issues.

good catch regarding the `ros-eloquent-gazebo*`, we seem to have targetted the wrong package as `exec_depend` in our `demos` package, https://github.com/osrf/rmf_demos/blob/master/demos/package.xml#L21, a fix is incoming after I run some tests

**Comment by aaronchongth** [comment]
The fix has been merged, please check before closing the issue.

**Comment by prashantking** [comment]
> Hi, sure aaron. After i install and run the office.launch.xml, the error is this:
> 
> ```
> kaeyan@kaeyan1804:~/rmf_demos_ws$ ros2 launch demos office.launch.xml
> [INFO] [launch]: All log files can be found below /home/kaeyan/.ros/log/2020-04-02-15-27-48-892762-kaeyan1804-6908
> [INFO] [launch]: Default logging verbosity is set to INFO
> [INFO] [rmf_traffic_schedule-1]: process started with pid [6919]
> [INFO] [building_map_server-2]: process started with pid [6920]
> [INFO] [rviz2-3]: process started with pid [6921]
> [INFO] [fleet_state_visualizer-4]: process started with pid [6922]
> [INFO] [rviz2-5]: process started with pid [6923]
> [INFO] [door_supervisor-6]: process started with pid [6924]
> [INFO] [gzserver-7]: process started with pid [6931]
> [INFO] [gzclient-8]: process started with pid [6932]
> [INFO] [magni_fleet_adapter-9]: process started with pid [6942]
> [INFO] [magni_state_aggregator-10]: process started with pid [6954]
> [INFO] [dispenser-11]: process started with pid [6957]
> [INFO] [dispenser-12]: process started with pid [6958]
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Launching fleet adapter for [magni]
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [linear_velocity] set to: 0.400000
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [angular_velocity] set to: 2.200000
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [linear_acceleration] set to: 3.000000
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [angular_acceleration] set to: 5.000000
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [profile_radius] set to: 0.300000
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [reversible] set to: 1
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [delay_threshold] set to: 15.000000
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [retry_wait] set to: 10.000000
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [planning_timeout] set to: 5.000000
> [magni_fleet_adapter-9] Adding waypoint [0] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [3] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [5] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [7] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [10] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [13] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [14] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [15] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [16] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [17] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [18] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [19] as a parking spot
> [magni_fleet_adapter-9] Adding waypoint [23] as a parking spot
> [magni_fleet_adapter-9] Named waypoints:
> [magni_fleet_adapter-9] -- [supplies]
> [magni_fleet_adapter-9] -- [magni2_charger]
> [magni_fleet_adapter-9] -- [coe]
> [magni_fleet_adapter-9] -- [magni1_charger]
> [magni_fleet_adapter-9] -- [hardware_2]
> [magni_fleet_adapter-9] -- [cubicle_2]
> [magni_fleet_adapter-9] -- [pantry]
> [magni_fleet_adapter-9] -- [station_1]
> [magni_fleet_adapter-9] -- [lounge]
> [magni_fleet_adapter-9] -- [cubicle_1]
> [magni_fleet_adapter-9] -- [hardware_1]
> [magni_fleet_adapter-9] -- [station_2]
> [magni_fleet_adapter-9] [INFO] [magni_fleet_adapter]: Parameter [discovery_timeout] set to: 60.000000
> [magni_fleet_adapter-9] Starting [magni]
> [gzclient-8] Gazebo multi-robot simulator, version 9.0.0
> [gzclient-8] Copyright (C) 2012 Open Source Robotics Foundation.
> [gzclient-8] Released under the Apache 2 License.
> [gzclient-8] http://gazebosim.org
> [gzclient-8]
> [gzserver-7] Gazebo multi-robot simulator, version 9.0.0
> [gzserver-7] Copyright (C) 2012 Open Source Robotics Foundation.
> [gzserver-7] Released under the Apache 2 License.
> [gzserver-7] http://gazebosim.org
> [gzserver-7]
> [gzserver-7] terminate called after throwing an instance of 'std::length_error'
> [gzserver-7] what(): basic_string::_M_create
> [gzclient-8] [Msg] Waiting for master.
> [gzclient-8] [Msg] Connected to gazebo master @ http://127.0.0.1:11345
> [gzclient-8] [Msg] Publicized address: 192.168.21.197
> [gzclient-8] [Err] [Node.cc:105] No namespace found
> [ERROR] [gzserver-7]: process has died [pid 6931, exit code -6, cmd 'gzserver --verbose -s libgazebo_ros_factory.so -s libgazebo_ros_init.so /home/kaeyan/rmf_demos_ws/install/rmf_demo_maps/share/rmf_demo_maps/maps/office/office.world'].
> [gzclient-8] [Err] [TransportIface.cc:396] Unable to connect to master.
> [gzclient-8] [Err] [GLWidget.cc:881] Unable to connect to a running Gazebo master.
> [gzclient-8] [Wrn] [Publisher.cc:141] Queue limit reached for topic /gazebo/world/user_camera/pose, deleting message. This warning is printed only once.
> ```

```
dpkg -l | grep gazebo9
```
what is the output of this ?

**Comment by aaronchongth** [comment]
that indicates that your build is missing the binaries that will come with this fix https://github.com/osrf/rmf_demos/pull/65, please start a new build with `rosdep` to pull in all the missing dependencies.
